### PR TITLE
author crud working

### PR DESCRIPTION
**Pull request by allistergg**

### CODE REVIEW

File: src/app/author/author.component.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        <label for="name">Name</label>
+      </div>
+      <div class="ui-g-8">
+        <input pInputText id="name" [(ngModel)]="author.name" autofocus />
+      </div>
+    </div>
+  </div>
+  <p-footer>
+    <div class="ui-dialog-buttonpane ui-helper-clearfix">
+      <button type="button" pButton icon="pi pi-times" (click)="delete()" label="Delete"></button>
```
Comment (sbhuju61) [inline src/app/author/author.component.html]: Add a if statement to hide delete button when adding author

File: src/app/author/author.component.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+
+    this.cols = [
+      { field: 'name', header: 'Name' },
+    ];
+  }
+
+  showDialogToAdd() {
+    this.newAuthor = true;
+    this.author = new PrimeAuthor();
```
Comment (sbhuju61) [inline src/app/author/author.component.ts]: empty object might be sufficient

Comment (sbhuju61) [reply src/app/author/author.component.ts]: this.author = {}

File: src/app/author/author.component.html (hunk 1 above)
Comment (sbhuju61) [reply src/app/author/author.component.html]: <button *ngIf="!newAuthor" type="button" pButton icon="pi pi-times" (click)="delete()" label="Delete"></button>

File: src/app/author/author.component.ts (hunk 2 above)
Comment (Jindanupajit-SS) [reply src/app/author/author.component.ts]: Empty object will work only if all members are optional

**Pull request merged**